common/djangoapps/student/models.py

```diff
@@ -166,9 +169,18 @@ def anonymous_id_for_user(user, course_id, save=True):
     if cached_id is not None:
         return cached_id
 
-    # include the secret key as a salt, and to make the ids unique across different LMS installs.
+    # check if an annonymous id already exists for this user and course_id combination
+    anonymous_user_ids = AnonymousUserId.objects.filter(user=user).filter(course_id=course_id)
+    if anonymous_user_ids:
+        if len(anonymous_user_ids) == 1:
+            return getattr(anonymous_user_ids[0], "anonymous_user_id")
+        else:
+            # there should only be one anonymous_user_id per user/course_id pair
+            raise Exception('Expected only one anonymous_user_id for user/course_id pair, instead ther are {}'.format(len(anonymous_user_ids)))
```

Comment:
This is moving away from a decision made earlier: https://github.com/edx/edx-platform/pull/13717 

We should decide what we want this to be and probably write up an ADR for this.